pkg/operator/staticpodoperator/controlloops/deployment_controller.go

```diff
@@ -0,0 +1,257 @@
+func (c DeploymentController) isLatestDeploymentCurrent(deploymentID int32) (bool, string) {
+	for _, name := range c.configMaps {
+		required, err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).Get(name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, err.Error()
+		}
+		existing, err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).Get(nameFor(name, deploymentID), metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, err.Error()
+		}
+		if !equality.Semantic.DeepEqual(existing.Data, required.Data) {
+			return false, fmt.Sprintf("configmap/%s has changed", required.Name)
+		}
+	}
+	for _, name := range c.secrets {
+		required, err := c.kubeClient.CoreV1().Secrets(c.targetNamespace).Get(name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, err.Error()
+		}
+		existing, err := c.kubeClient.CoreV1().Secrets(c.targetNamespace).Get(nameFor(name, deploymentID), metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, err.Error()
+		}
+		if !equality.Semantic.DeepEqual(existing.Data, required.Data) {
+			return false, fmt.Sprintf("secret/%s has changed", required.Name)
+		}
+	}
+
+	return true, ""
+}
+
+func (c DeploymentController) createNewDeploymentController(deploymentID int32) error {
+	for _, name := range c.configMaps {
+		obj, _, err := resourceapply.SyncConfigMap(c.kubeClient.CoreV1(), c.targetNamespace, name, c.targetNamespace, nameFor(name, deploymentID))
+		if err != nil {
+			return err
+		}
+		if obj == nil {
+			return apierrors.NewNotFound(corev1.Resource("configmaps"), name)
+		}
+	}
+	for _, name := range c.secrets {
+		obj, _, err := resourceapply.SyncSecret(c.kubeClient.CoreV1(), c.targetNamespace, name, c.targetNamespace, nameFor(name, deploymentID))
+		if err != nil {
+			return err
+		}
+		if obj == nil {
+			return apierrors.NewNotFound(corev1.Resource("secrets"), name)
+		}
+	}
+
+	return nil
+}
+
+func (c DeploymentController) sync() error {
+	operatorSpec, originalOperatorStatus, resourceVersion, err := c.operatorConfigClient.Get()
+	if err != nil {
+		return err
+	}
+	operatorStatus := originalOperatorStatus.DeepCopy()
+
+	switch operatorSpec.ManagementState {
+	case operatorv1alpha1.Unmanaged:
+		return nil
+	case operatorv1alpha1.Removed:
+		// TODO probably just fail.  Static pod managers can't be removed.
+		return nil
+	}
+
+	requeue, syncErr := c.createDeploymentController(operatorSpec, operatorStatus, resourceVersion)
+	if requeue && syncErr == nil {
+		return fmt.Errorf("synthetic requeue request")
+	}
+	err = syncErr
+
+	if err != nil {
+		v1alpha1helpers.SetOperatorCondition(&operatorStatus.Conditions, operatorv1alpha1.OperatorCondition{
+			Type:    operatorv1alpha1.OperatorStatusTypeFailing,
+			Status:  operatorv1alpha1.ConditionTrue,
+			Reason:  "StatusUpdateError",
+			Message: err.Error(),
+		})
+		if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
```

Comment:
where do we mutate the `operatorStatus` ? will this code ever execute? i think the intention was to report https://github.com/openshift/library-go/pull/61/files#diff-5af6e2a63b1f0f33b0fc2484dfa041b3R192 ?